// DO NOT EDIT! This test has been generated by /html/canvas/tools/gentest.py.
// OffscreenCanvas test in a worker:2d.layer.malformed-operations.getImageData
// Description:Check that exceptions are thrown for operations that are malformed while layers are open.
// Note:

importScripts("/resources/testharness.js");
importScripts("/html/canvas/resources/canvas-tests.js");

var t = async_test("Check that exceptions are thrown for operations that are malformed while layers are open.");
var t_pass = t.done.bind(t);
var t_fail = t.step_func(function(reason) {
    throw reason;
});
t.step(function() {

  var canvas = new OffscreenCanvas(200, 200);
  var ctx = canvas.getContext('2d');

  // Shouldn't throw on its own.
  ctx.getImageData(0, 0, 200, 200);
  // Make sure the exception isn't caused by calling the function twice.
  ctx.getImageData(0, 0, 200, 200);
  // Calling again inside a layer should throw.
  ctx.beginLayer();
  assert_throws_dom("InvalidStateError",
                    () => ctx.getImageData(0, 0, 200, 200));
  t.done();
});
done();
